lnwire/writer.go

```diff
@@ -0,0 +1,445 @@
+package lnwire
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"image/color"
+	"math"
+	"net"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcutil"
+	"github.com/lightningnetwork/lnd/tor"
+)
+
+var (
+	// ErrNilFeatureVector is returned when the supplied feature is nil.
+	ErrNilFeatureVector = errors.New("cannot write nil feature vector")
+
+	// ErrPkScriptTooLong is returned when the length of the provided
+	// script exceeds 34.
+	ErrPkScriptTooLong = errors.New("'PkScript' too long")
+
+	// ErrNilTCPAddress is returned when the supplied address is nil.
+	ErrNilTCPAddress = errors.New("cannot write nil TCPAddr")
+
+	// ErrNilOnionAddress is returned when the supplied address is nil.
+	ErrNilOnionAddress = errors.New("cannot write nil onion address")
+
+	// ErrNilNetAddress is returned when a nil value is used in []net.Addr.
+	ErrNilNetAddress = errors.New("cannot write nil address")
+
+	// ErrNilPublicKey is returned when a nil pubkey is used.
+	ErrNilPublicKey = errors.New("cannot write nil pubkey")
+
+	// ErrUnknownServiceLength is returned when the onion service length is
+	// unknown.
+	ErrUnknownServiceLength = errors.New("unknown onion service length")
+)
+
+// ErrOutpointIndexTooBig is used when the outpoint index exceeds the max value
+// of uint16.
+func ErrOutpointIndexTooBig(index uint32) error {
+	return fmt.Errorf(
```

Comment:
error string on first line?

Reply:
I think it only applies to log messages.